### PR TITLE
✨ Add buttons row

### PR DESCRIPTION
**Pull request by iantrich**
## Proposed change

Add `buttons` row for entities card

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: entities
entities:
  - type: buttons
    entities:
      - light.bed_light
      - entity: light.kitchen_lights
        icon: 'mdi:lightbulb-group'
```

## Additional information

 
image
top is the header buttons. middle is the buttons row.

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/components/hui-buttons-base.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  protected render(): TemplateResult | void {
+    return html`
+      ${(this._configEntities || []).map((entityConf) => {
+        const stateObj = this._hass!.states[entityConf.entity];
+        if (!stateObj) {
+          return html`<div class='missing'><iron-icon icon="hass:alert"></div>`;
+        }
+
+        return html`
+          <div>
```
Comment (balloob) [inline src/panels/lovelace/components/hui-buttons-base.ts]: Should it be in a div or should we add `:host {style: block; }`

File: src/panels/lovelace/components/hui-buttons-base.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+// tslint:disable-next-line: no-duplicate-imports
+import { StateBadge } from "../../../components/entity/state-badge";
+
+@customElement("hui-buttons-base")
+export class HuiButtonsBase extends LitElement {
+  public static getStubConfig(): object {
+    return { entities: [] };
+  }
+
+  protected _configEntities?: EntityConfig[];
```
Comment (balloob) [inline src/panels/lovelace/components/hui-buttons-base.ts]: Why not make this a property?

File: src/panels/lovelace/components/hui-buttons-base.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import "../../../components/ha-icon";
+
+import { HomeAssistant } from "../../../types";
+import { EntityConfig } from "../entity-rows/types";
+import { toggleEntity } from "../common/entity/toggle-entity";
+import { computeTooltip } from "../common/compute-tooltip";
+// tslint:disable-next-line: no-duplicate-imports
+import { StateBadge } from "../../../components/entity/state-badge";
+
+@customElement("hui-buttons-base")
```
Comment (balloob) [inline src/panels/lovelace/components/hui-buttons-base.ts]: Why register this element if you just inherit from it ?

File: src/panels/lovelace/special-rows/hui-buttons-row.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { customElement } from "lit-element";
+
+import { HuiButtonsBase } from "../components/hui-buttons-base";
+import { LovelaceRow, ButtonsRowConfig } from "../entity-rows/types";
+import { processConfigEntities } from "../common/process-config-entities";
+
+@customElement("hui-buttons-row")
```
Comment (balloob) [inline src/panels/lovelace/special-rows/hui-buttons-row.ts]: Will the buttons row need extra padding ?

File: src/panels/lovelace/header-footer/hui-buttons-header-footer.ts [hunk 5]
```diff
@@ -1,98 +1,17 @@
...
-import { EntityConfig } from "../entity-rows/types";
 import { processConfigEntities } from "../common/process-config-entities";
-import { toggleEntity } from "../common/entity/toggle-entity";
-import { computeTooltip } from "../common/compute-tooltip";
-// tslint:disable-next-line: no-duplicate-imports
-import { StateBadge } from "../../../components/entity/state-badge";
 
 @customElement("hui-buttons-header-footer")
-export class HuiButtonsHeaderFooter extends LitElement
+export class HuiButtonsHeaderFooter extends HuiButtonsBase
```
Comment (balloob) [inline src/panels/lovelace/header-footer/hui-buttons-header-footer.ts]: I prefer composition over inheritance. Why not render `<hui-button-base>` ?

File: src/panels/lovelace/special-rows/hui-buttons-row.ts (hunk 4 above)
Comment (iantrich) [reply src/panels/lovelace/special-rows/hui-buttons-row.ts]: No, I think it looked fine. I'll add a screenshot in a bit

File: src/panels/lovelace/components/hui-buttons-base.ts (hunk 3 above)
Comment (iantrich) [reply src/panels/lovelace/components/hui-buttons-base.ts]: using composition now as suggestsed

File: src/panels/lovelace/create-element/create-row-element.ts [hunk 6]
```diff
@@ -15,6 +16,7 @@ const ALWAYS_LOADED_TYPES = new Set([
   "sensor-entity",
   "text-entity",
   "toggle-entity",
+  "buttons",
```
Comment (bramkragten) [inline src/panels/lovelace/create-element/create-row-element.ts]: Please move it to the lazy loaded

**Review by bramkragten — APPROVED** [review]

**Pull request merged**